web/html/src/manager/content-management/shared/components/messages/messages.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import * as React from 'react';
+import _isEmpty from "lodash/isEmpty";
+
+import {Messages} from 'components/messages';
+import type {ProjectMessageType} from '../../type/project.type';
+
+type ValidationMessagesType = {
+  panelClass: string,
+  messages: ?React.Element<'ul'>
+}
+
+const msgClassMap = {
+  'warning': {
+    text: 'text-warning',
+    panel: 'panel-warning'
+  },
+  'error': {
+    text: 'text-danger',
+    panel: 'panel-danger'
+  },
+  'info': {
+    text: 'text-default',
+    panel: 'panel-default'
+  }
+}
+
+const sortMessages = (messages: Array<ProjectMessageType>) => {
+  const msgPriorities = {'error': 0, 'warning': 1, 'info': 2};
+  return messages.slice().sort((a, b) => msgPriorities[a.type] - msgPriorities[b.type]);
+}
+
+const getMessages = (messages: Array<ProjectMessageType>): ValidationMessagesType => {
+  if (_isEmpty(messages)) {
+    return {
+      panelClass: 'panel-default',
+      messages: null
+    }
+  }
+
+  const sortedMsgs = sortMessages(messages);
+  return {
+    panelClass: msgClassMap[sortedMsgs[0].type].panel,
+    messages: (
+      <Messages items={sortedMsgs.map(m => ({severity: m.type, text: m.text}))}/>
+    )
+  }
+}
+
+export default getMessages;
```

Comment:
I don't see any usage of this, only `getRenderedMessages`. A typo?

Reply:
Since it is the default export, you can import it with any arbitrary name. But since it's always imported as `getRenderedMessages`, I guess it makes sense to rename this as well :+1:

Reply:
I see, thanks for explaining. Renaming might clarify it a bit, as you say.